### PR TITLE
Classification

### PR DESCRIPTION
**Pull request by jxmorris12**
builds on  #7

### CODE REVIEW

File: dataloaders/helpers.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        return tuple(torch.stack(d[k]) for k in text_columns) + tuple(torch.stack(d[k]).float() for k in label_columns)
+    return DataLoader(dataset,
+        batch_size=batch_size, collate_fn=collate_fn,
+        shuffle=True, drop_last=drop_last, pin_memory=True
+    )
+
+
+def load_rotten_tomatoes(batch_size: int, max_length: int, drop_last: bool = True, num_examples: int = None) -> Tuple[DataLoader, DataLoader]:
+    """Returns tuple of (train_dataloader, test_dataloader)."""
+    dataset = datasets.load_dataset('rotten_tomatoes').shuffle(seed=42)
```
Comment (jscuds) [inline dataloaders/helpers.py]: Does it matter if `seed=42` or will it be overridden by `torch.manual_seed(args.random_seed)`?

File: dataloaders/helpers.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    if num_examples:
+        # For some reason, truncating datasets like this returns a plain dict,
+        # so need to call Dataset.from_dict to restore as a Dataset type.
+        for split in dataset:
+            dataset[split] = datasets.Dataset.from_dict(dataset[split][:num_examples])
```
Comment (jscuds) [inline dataloaders/helpers.py:97]: doesn't this force both `train` and `test`/`val` sets to be the same size?  Is that desirable?

File: experiment.py [hunk 3]
```diff
@@ -99,17 +95,53 @@ def __init__(self, args: argparse.Namespace):
...
 
     @property
     def M(self) -> torch.nn.Parameter:
         """The orthogonal matrix, M, used for retrofitting."""
-        if self.args.model_name_or_path == 'elmo':
+        if self.args.model_name == 'elmo_single_sentence':
             return self.model.elmo._elmo_lstm._elmo_lstm.M
         else:
-            raise ValueError(f'cannot get orthogonal matrix for model {self.model_name_or_path}')
+            raise ValueError(f'cannot get orthogonal matrix for model {self.args.model_name}')
```
Comment (jscuds) [inline experiment.py:123]: is this saying that for an `elmo_sentence_pair` task you won't be able to get the `M` matrix?

File: train.py [hunk 4]
```diff
@@ -101,15 +116,13 @@ def run_training_loop(args: argparse.Namespace):
...
     day = time.strftime(f'%Y-%m-%d-%H%M')
-    exp_name = f'{args.experiment}_{args.model_name_or_path}_{day}'
+    exp_name = f'{args.experiment}_{args.model_name}_{day}'
     # WandB init and config (based on argument dictionaries in imports/globals cell)
     wandb.init(
         name=exp_name,
-        project='rf-bert',
-        entity='jscuds',
+        project=os.environ.get('WANDB_PROJECT', 'rf-bert'),
+        entity=os.environ.get('WANDB_ENTITY', 'jscuds'),
```
Comment (jscuds) [inline train.py:125]: This pulls the corresponding environment variable, and if it's not set to anything defaults to `'rf-bert'` and `'jscuds'`...correct?

File: experiment.py (hunk 3 above)
Comment (jxmorris12) [reply experiment.py:123]: Yeah this is a good question, it's not ideal. I just have this here so we can support non-ELMO models later.  I arbitrarily decided that the model has to be called 'elmo_single_sentence' which just means ELMO when the experiment is 'retrofit'. It's not even really correct since retrofitting uses two sentences (actually four?) 

but yeah if there's a better way you can think of let me know. we could make the argparse take three models: `{'elmo_retrofit', 'elmo_single_sentence', 'elmo_sentence_pair'}` but that seems ugly too.

File: dataloaders/helpers.py (hunk 2 above)
Comment (jxmorris12) [reply dataloaders/helpers.py:97]: yes another great question!
1. Typically we shouldn't use this argument. We usually want to use the full train/val set
2. when this argument is set, it only sets the maximum length, so if the test set is 1000 examples and num_examples is 20k, it won't remove any examples from the test set

File: train.py (hunk 4 above)
Comment (jxmorris12) [reply train.py:125]: yeah exactly. this is so I can type `WANDB_ENTITY='jxmorris12' python train.py` because otherwise it'll try to connect me to your account and break

File: dataloaders/helpers.py (hunk 1 above)
Comment (jxmorris12) [reply dataloaders/helpers.py]: Good question. I don't think it'll make a difference in practice but we should probably pipe in the random seed. I'll fix that

File: experiment.py (hunk 3 above)
Comment (jscuds) [reply experiment.py:123]: Makes sense - this seems good for now.  I didn't make the connection that it defaults to `'elmo_single_sentence'` under the `retrofit` scenario.

File: dataloaders/helpers.py (hunk 1 above)
Comment (jxmorris12) [reply dataloaders/helpers.py]: Ok fixed
